### DISCUSSION TITLE
Temperature-dependent adjustment of the input geometry

### DISCUSSION THREAD

**Discussion by sag-kalibreur** [pyroll-project/pyroll-core#220] [Q&A] [answered]
### Background

It is often necessary to calculate the expansion of the geometry of the inprofile with a linear expansion coefficient or similar due to different input temperatures.

### `Question`

What is the best way to implement this calculation so that it automatically adjusts the input geometry? 

### First Idea

A first idea would be to do this using the mean linear expansion coefficient (constant or as a temperature-dependent table (with reference temperature) in a material data file.

###  Your opinion is welcome

Please share your ideas or write a comment..

**Comment by RollingTechSupport** [comment]
We have a plugin prepared called `linear-thermal-expansion` that should fit your needs. 
Will be tested and added to pyroll-basic soon.

**Reply by sag-kalibreur** [reply]
That sounds good, I'm already looking forward to the realisation...

**Reply by RollingTechSupport** [reply, marked as answer]
So I created the issue #224 that will deal with the implementation. As for `Transport`, the plugin uses the  coefficient of thermal expansion to calculate the temperature dependent cross-section. Inside a `RollPass` only the length of the workpiece will be changed.
